docs/developer/proposal/EP_004-ACPI-support.md

```diff
@@ -0,0 +1,292 @@
+
+Add ACPI option which is an alternative of Redfish to Kepler.
+ACPI can collect platform power metrics from reading sysfs. The platform collector is able to reuse the one that Redfish uses if merged.
+
+```text
+┌─────────────────┐    ┌───────────────────┐
+│   CPU Power     │    │  Platform Power   │
+│   (RAPL)        │    │ (Redfish or ACPI) │
+└─────────────────┘    └───────────────────┘
+         │                        │
+         ▼                        ▼
+┌─────────────────┐    ┌────────────────────┐
+│ Power Monitor   │    │ Power Monitor      │
+│ (Attribution)   │    │ (For ACPI,         │
+│                 │    │ no attribution)    │
+└─────────────────┘    └────────────────────┘
+         │                        │
+         ▼                        ▼
+┌─────────────────┐    ┌────────────────────┐
+│ Power Collector │    │ Platform Collector │
+│                 │    │                    │
+└─────────────────┘    └────────────────────┘
+         └──────────┬─────────────┘
+                    ▼
+           ┌──────────────────┐
+           │ Prometheus       │
+           │ Exporter         │
+           └──────────────────┘
+```
+
+### Aggregate power metrics
+
+According to [the ACPI spec for power meter](https://docs.kernel.org/hwmon/acpi_power_meter.html#special-features), there may be more than one power meter available per platform (`/sys/bus/acpi/drivers/power_meter/ACPI000D:XX/power[1-*]_average`).
+To acquire power metrics for whole platform, I will aggregate metrics from all available power meters.
+
+## Detailed Design
+
+### Package Structure
+
+```text
+cmd/
+├── kepler/
+│   └── main.go  # Add ACPI power monitor when enabled
+config/
+│   └── config.go  # Add config for ACPI
+internal/
+├── device/
+│   └── platform/
+│       └── acpi/
+|           ├── testdata/
+|           |   └── sys/bus/acpi/drivers/power_meter/...
+│           ├── service.go  # Power monitor for ACPI implementation
+│           ├── acpi_power_meter_test.go
+│           └── acpi_power_meter.go  # Implements power_meter interface which read sysfs to get metrics
+└── exporter/prometheus/collector/
+    └── platform_collector.go  # Platform power metrics collector
+```
+
+### API/Interface Changes
+
+[Describe any changes to public APIs, service interfaces, or data structures. Show code snippets for new or modified interfaces.]
+
+```go
+// acpi_power_meter.go - Implements powerMeter interface
+type acpiPowerMeter struct {
+    logger *slog.Logger
+    devicePath string
+}
+
+// Service for ACPI power meter
+type Service struct {
+    logger *slog.Logger
+    powerMeter acpiPowerMeter
+}
+```
+
+## Configuration
+
+### Main Configuration Changes
+
+`ACPI.Enabled` is False by default.
+
+```go
+type Platform struct {
+    Redfish Redfish `yaml:"redfish"`
+    ACPI    ACPI    `yaml:"acpi"`
+}
+
+type ACPI struct {
+    Enabled    *bool  `yaml:"enabled"`
+}
+```
+
+**CLI flags**
+
+```
+--platform.acpi.enabled=true  # Enable ACPI monitoring
```

Comment:
Please note that for the actual feature implementation, lets introduce this as an `experimental` feature, get feedback and then promote it as a supported feature.